irohad/torii/processor/impl/stub_query_processor.cpp

```diff
@@ -25,7 +25,9 @@ namespace iroha {
     using dao::QueryResponse;
     using dao::Client;
 
-    QueryProcessorStub::QueryProcessorStub() {
+    QueryProcessorStub::QueryProcessorStub(ametsuchi::WsvQuery &wsv,
+                                           ametsuchi::BlockQuery &block) :
```

Comment:
> Ametsuchi is a highly coupled component - it will be a problem to answer query such as getWalletTransactions, if BlockQuery and WsvQuery are separate, so Ametsuchi implements both interfaces.

Thanks, I understand that.
So, it can be expressed:
```
QueryProcessorStub::QueryProcessorStub(AmetsuchiStub &ametsuchi) :
    wsv_(ametsuchi), block_(ametsuchi) {
```

in `irohad/main/iroha-main.cpp`
```
iroha::torii::QueryProcessorStub qp(ametsuchi);
```
Using ametsuchi 2 times to 1 time.